CBLClient/Apps/CBLTestServer-Android/app/src/main/java/com/couchbase/CouchbaseLiteServ/server/RequestHandler/LoggingRequestHandler.java

```diff
@@ -147,4 +151,23 @@ public LogFileConfiguration setLogLevel(Args args) {
         return config;
     }
 
+    public RawData getLogsInZip(Args args) {
+        LogFileConfiguration fileLoggerConfig = Database.log.getFile().getConfig();
+        if (fileLoggerConfig == null) { return null; }
+
+        ZipUtils zipper = new ZipUtils();
+
+        File zipDir = CouchbaseLiteServ.getAppContext().getExternalFilesDir("zip");
+        try {
+            File zipFile = new File(zipDir, "archive.zip");
+            if (zipFile.exists()) { zipper.deleteRecursive(zipFile); }
+
+            zipper.zipDirectory(fileLoggerConfig.getDirectory(), zipFile);
+
+            return new RawData("application/zip", zipper.readFile(zipFile));
```

Comment:
IDE didn't complain there is no return outside of the try{}? need to return null on exception even though the logic doesn't need to catch exception here.

Reply:
Why would it complain?  As far as I can see, this code is perfectly correct.  It will either return a new RawData, or throw.

Reply:
ok, I think I missed the throw part. sorry